recipe-server/client/control_new/components/recipes/RecipeForm.js

```diff
@@ -0,0 +1,128 @@
+import { Button, Form, Input, Select } from 'antd';
+import autobind from 'autobind-decorator';
+import { is, Map } from 'immutable';
+import pt from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { createForm, FormItem } from 'control_new/components/common/forms';
+import FormActions from 'control_new/components/common/FormActions';
+import ConsoleLogFields from 'control_new/components/recipes/ConsoleLogFields';
+import PreferenceExperimentFields from 'control_new/components/recipes/PreferenceExperimentFields';
+import ShowHeartbeatFields from 'control_new/components/recipes/ShowHeartbeatFields';
+import { getAction, getAllActions } from 'control_new/state/actions/selectors';
+
+
+/**
+ * Form for editing recipes.
+ */
+@createForm({})
+@connect(
+  (state, props) => ({
+    selectedAction: getAction(
+      state,
+      props.form.getFieldValue('action_id'),
+      new Map(),
+    ),
+  }),
+)
+@autobind
+export default class RecipeForm extends React.Component {
+  static propTypes = {
+    recipe: pt.instanceOf(Map).isRequired,
+    form: pt.object.isRequired,
+    onSubmit: pt.func.isRequired,
+    selectedAction: pt.instanceOf(Map).isRequired,
+  }
+
+  static argumentsFields = {
+    'console-log': ConsoleLogFields,
+    'show-heartbeat': ShowHeartbeatFields,
+    'preference-experiment': PreferenceExperimentFields,
+  };
+
+  componentWillReceiveProps(newProps) {
+    // Initial values are mostly handled via props, but if the recipe
+    // changes, we need to reset the values stored in the state.
+    if (!is(newProps.recipe, this.props.recipe)) {
+      this.props.form.resetFields();
+    }
+  }
+
+  render() {
+    const { onSubmit, recipe = new Map(), selectedAction } = this.props;
+    const ArgumentsFields = RecipeForm.argumentsFields[selectedAction.get('name')];
+
+    return (
+      <Form onSubmit={onSubmit}>
+        <FormItem
+          name="name"
+          label="Name"
+          initialValue={recipe.get('name')}
+        >
+          <Input />
+        </FormItem>
+        <FormItem
+          name="extra_filter_expression"
+          label="Filter Expression"
+          initialValue={recipe.get('extra_filter_expression')}
+        >
+          <Input type="textarea" rows="4" />
+        </FormItem>
+        <FormItem
+          name="action_id"
+          label="Action"
+          initialValue={recipe.getIn(['action', 'id']).toString(10)}
```

Comment:
Nope! Looking into it, the `ActionSelect` component already `toString`s the value, so we don't need to do so here. I've updated the `ActionSelect`'s propTypes to resolve the console warning.